src/tainted/transaction.h

```diff
@@ -71,6 +72,45 @@ class Transaction {
         }
     }
 
+    bool HasJsObjectReference() const noexcept {
+        return !_jsObjectRef.IsEmpty();
+    }
+
+    transaction_key_t GetCurrentTransactionKey() const noexcept {
+        if (_jsObjectRef.IsEmpty()) {
+            return _id;
+        }
+        auto isolate = v8::Isolate::GetCurrent();
+        if (!isolate) {
+            return _id;
+        }
+        v8::HandleScope handle_scope(isolate);
```

Comment:
Isn't this equivalent of the next block?
```
v8::HandleScope handle_scope = v8::HandleScope(isolate);
```

You are not using `handle_scope` variable, why is this needed?

Reply:
> Isn't this equivalent of the next block?

No, it is not. Your proposal is a _copy initialization_ (creates a temp object and then moves it) meanwhile the current form is a _direct construction_ . The first one requires the class to be copyable or movable, and `v8::HandleScope` is not, due to its internals.

> You are not using handle_scope variable, why is this needed?

It is true that it is not used per se, but it is needed to manage the `v8::Local<v8::Value>` that is created on `_jsObjectRef.Get(isolate)` a bit later.